pkg/executor/join/hash_join_v2.go

```diff
@@ -588,40 +524,31 @@ func (w *ProbeWorkerV2) getNewJoinResult() (bool, *hashjoinWorkerResult) {
 func (e *HashJoinV2Exec) Next(ctx context.Context, req *chunk.Chunk) (err error) {
 	if !e.prepared {
 		e.initHashTableContext()
+		e.initializeForProbe()
```

Comment:
why move `e.initializeForProbe()` out from `e.fetchAndProbeHashTable()`?

Reply:
> why move `e.initializeForProbe()` out from `e.fetchAndProbeHashTable()`?

Because `joinResultCh` in `e.initializeForProbe()` function needs to be initialized before starting build fetcher, as `joinResultCh` is also used for transmitting error.